server/src/main/java/org/elasticsearch/common/collect/ImmutableOpenMap.java

```diff
@@ -136,6 +136,23 @@ public void remove() {
         };
     }
 
+    /**
+     * Returns a direct stream over the keys.
+     */
+    public Stream<KType> keysStream() {
```

Comment:
Instead of add another non-standard method, can we expose the keyset? This would allow operating on ImmutableOpenMap just like a java Map: get the keyset and call stream() on it.

Reply:
I think exposing `keySet` would be difficult without additional copying. Exposing a `Stream` allows the consumer to iterate over the keys without copying them to a set. Also, we already added the `stream` method #76921, so I think it would be consistent to have `keyStream` as alternative to `keyIt`.

Reply:
I don't think exposing a keySet would be much different than the keys() method on the underlying ObjectObjectHashMap. The set doesn't need to be materialized, it is just a wrapper around the keys() object.